uvloop/loop.pyi

```diff
@@ -0,0 +1,397 @@
+    def writelines(self, list_of_data: List[Any]) -> None: ...
+    def write_eof(self) -> None: ...
+    def can_write_eof(self) -> bool: ...
+    def pause_reading(self) -> None: ...
+    def resume_reading(self) -> None: ...
+
+class _UVStreamServer(_UVSocketHandle): ...
+class _TCPTransport(_UVStream): ...
+class _UDPTransport(_UVBaseTransport):
+    def sendto(self, data: Any, addr: Any = ...) -> None: ...
+
+class _UnixServer(_UVStreamServer, asyncio.AbstractServer): ...
+class _UnixTransport(_UVStream): ...
+class _ReadUnixTransport(_UVStream):
+    def get_write_buffer_limits(self) -> Tuple[int, int]: ...
+    def set_write_buffer_limits(self, high: Optional[int] = ..., low: Optional[int] = ...) -> None: ...
+    def get_write_buffer_size(self) -> int: ...
+    def write(self, data: Any) -> None: ...
+    def writelines(self, list_of_data: List[Any]) -> None: ...
+    def write_eof(self) -> None: ...
+    def can_write_eof(self) -> bool: ...
+    def abort(self) -> None: ...
+
+class _WriteUnixTransport(_UVStream):
+    def pause_reading(self) -> None: ...
+    def resume_reading(self) -> None: ...
+
+class _UVProcess(_UVHandle): ...
+class _UVProcessTransport(_UVProcess, asyncio.BaseTransport):
+    def get_protocol(self) -> asyncio.BaseProtocol: ...
+    def set_protocol(self, protocol: asyncio.BaseProtocol) -> None: ...
+    def get_pid(self) -> Optional[int]: ...
+    def get_returncode(self) -> Optional[int]: ...
+    def get_pipe_transport(self, fd: int) -> Optional[
+        Union[_WriteUnixTransport, _ReadUnixTransport]
+    ]: ...
+    def terminate(self) -> None: ...
+    def kill(self) -> None: ...
+    def send_signal(self, signal: int) -> None: ...
+    def is_closing(self) -> bool: ...
+    def close(self) -> None: ...
+    def get_extra_info(self, name: Any, default: Any = ...) -> Any: ...
+
+class _SSLProtocolTransport(asyncio.BaseTransport):
+    def get_extra_info(self, name: Any, default: Any = ...) -> Any: ...
+    def set_protocol(self, protocol: asyncio.BaseProtocol) -> None: ...
+    def get_protocol(self) -> asyncio.BaseProtocol: ...
+    def is_closing(self) -> bool: ...
+    def close(self) -> None: ...
+    def is_reading(self) -> bool: ...
+    def pause_reading(self) -> None: ...
+    def resume_reading(self) -> None: ...
+    def set_write_buffer_limits(self, high: Optional[int] = ..., low: Optional[int] = ...) -> None: ...
+    def get_write_buffer_limits(self) -> Tuple[int, int]: ...
+    def get_write_buffer_size(self) -> int: ...
+    def set_read_buffer_limits(self, high: Optional[int] = ..., low: Optional[int] = ...) -> None: ...
+    def get_read_buffer_limits(self) -> Tuple[int, int]: ...
+    def get_read_buffer_size(self) -> int: ...
+    def write(self, data: Any) -> None: ...
+    def writelines(self, list_of_data: List[Any]) -> None: ...
+    def write_eof(self) -> None: ...
+    def can_write_eof(self) -> bool: ...
+    def abort(self) -> None: ...
+
+class _Server(asyncio.AbstractServer):
```

Comment:
I'm curious why we need this as opposed to simply using `asyncio.AbstractServer`?

Reply:
`asyncio.AbstractServer` is defined as follows in typeshed:

```python
class AbstractServer:
    sockets: Optional[List[socket]]
    def close(self) -> None: ...
    if sys.version_info >= (3, 7):
        async def __aenter__(self: _T) -> _T: ...
        async def __aexit__(self, *exc: Any) -> None: ...
        def get_loop(self) -> AbstractEventLoop: ...
        def is_serving(self) -> bool: ...
        async def start_serving(self) -> None: ...
        async def serve_forever(self) -> None: ...
    async def wait_closed(self) -> None: ...
```

Since `uvloop` works on Python 3.5+ and `Server` defines those methods for all versions, `asyncio.AbstractServer` will be wrong for `uvloop` on Python 3.5 and 3.6.

Reply:
I'll be dropping support for 3.5 and 3.6. 3.5 is no longer supported by Python core devs; 3.6 is in security fixes only mode. And with 3.9 now out, uvloop will be 3.7+ only in its next release, supporting three Python versions: 3.7, 3.8, and 3.9.

Reply:
That's good to know! I'll go back over this PR with that in mind.

Reply:
One thing to note: I think we'll still need `_Server` because `AbstractServer.sockets` is `Optional[List[Socket]]` and `Server.sockets` is `List[Socket]`.

Reply:
I think we should stick to AbstractServer typing here too. That makes it easy for us to fix our `Server.sockets` property to follow the spec.

Reply:
Some of these others I added while I was navigating the source. I'll make sure I validate all of these are needed when I go over this again.

Reply:
Yes, thanks. And we can always fix uvloop code that doesn't conform to asyncio APIs.

Reply:
Sorry that it's taken so long to get back to this. I've updated the PR based on your feedback and rebased on master. Unfortunately, the PR checks are failing prematurely.